### PR TITLE
Add inline documentation for Decoded

### PR DESCRIPTION
**Pull request by gfontenot**
Also a small organizational style tweak while I was in there.

### CODE REVIEW

File: Argo/Types/Decoded/Decoded.swift [hunk 1]
```diff
@@ -28,6 +56,14 @@ public extension Decoded {
     }
   }
 
+  /**
+    Convert an `Optional` into a `Decoded` value.
+
+    If the provided optional is `.Some`, this method extracts the value and
+    wraps it in `.Success`. Otherwise, it returns a `.TypeMismatch` error
```
Comment (gabebw) [inline Argo/Types/Decoded/Decoded.swift]: You probably want a period at the end of this line.

File: Argo/Types/Decoded/Decoded.swift [hunk 2]
```diff
@@ -28,6 +56,14 @@ public extension Decoded {
...
+
+    - returns: The provided `Optional` value transformed into a `Decoded` value
```
Comment (gabebw) [inline Argo/Types/Decoded/Decoded.swift]: Period at the end of this line, to match the rest of the docs.

Comment (gabebw) [inline Argo/Types/Decoded/Decoded.swift]: Oh actually none of the `parameter` lines end in a period and of the `returns` docs, only line 83 ends in a period. So this is fine, but you may want to change line 83.

File: Argo/Types/Decoded/Decoded.swift [hunk 3]
```diff
@@ -1,15 +1,28 @@
+/// The result of a failable decoding operation
```
Comment (tonyd256) [inline Argo/Types/Decoded/Decoded.swift]: period?

File: Argo/Types/Decoded/Decoded.swift [hunk 4]
```diff
@@ -37,14 +73,39 @@ public extension Decoded {
...
 
 public extension Decoded {
+  /**
+    Convenience function for creating `.TypeMismatch` errors.
+
+    - parameter expected: The expected type
+    - parameter actual: The actual type
+
+    - returns: A `Decoded.Failure` with a `.TypeMismatch` error constructed
+               from the provided `expected` and `actual` values.
```
Comment (tonyd256) [inline Argo/Types/Decoded/Decoded.swift]: remove `.`

File: Argo/Types/Decoded/Decoded.swift [hunk 5]
```diff
@@ -37,14 +73,39 @@ public extension Decoded {
...
     return .Failure(.MissingKey(name))
   }
 
+  /**
+    Convenience function for creating `.Custom` errors
+
+    - parameter message: The custom error message
+
+    - returns: A `Decoded.Failure` with a `.Custom` error constructed from the
+               provided `message` value.
```
Comment (tonyd256) [inline Argo/Types/Decoded/Decoded.swift]: remove `.`

File: Argo/Types/Decoded/Decoded.swift [hunk 6]
```diff
@@ -60,6 +121,17 @@ extension Decoded: CustomStringConvertible {
 }
 
 public extension Decoded {
+  /**
+    Extract the success value or throw an error.
```
Comment (tonyd256) [inline Argo/Types/Decoded/Decoded.swift]: `.Success`?

File: Argo/Types/Decoded/Decoded.swift [hunk 7]
```diff
@@ -60,6 +121,17 @@ extension Decoded: CustomStringConvertible {
...
+
+    This can be used to move from `Decoded` types into the world of `throws`.
+    If the value exists, this will return it. Otherwise, it will throw the error
+    information.
+
+    - throws: `DecodeError` if `self` is `.Failure`
+
+    - returns: The unwrapped success case
```
Comment (tonyd256) [inline Argo/Types/Decoded/Decoded.swift]: "The unwrapped value"?

File: Argo/Types/Decoded/Decoded.swift [hunk 8]
```diff
@@ -37,14 +73,39 @@ public extension Decoded {
 }
 
 public extension Decoded {
+  /**
+    Convenience function for creating `.TypeMismatch` errors.
+
+    - parameter expected: The expected type
```
Comment (klaaspieter) [inline Argo/Types/Decoded/Decoded.swift]: Considering `expected` is a `String` that reflects a type, some guidance on what to put here might be helpful.

File: Argo/Types/Decoded/Decoded.swift [hunk 9]
```diff
@@ -37,14 +73,39 @@ public extension Decoded {
...
   static func typeMismatch<T, U>(expected: String, actual: U) -> Decoded<T> {
     return .Failure(.TypeMismatch(expected: expected, actual: String(actual)))
   }
 
+  /**
+    Convenience function for creating `.MissingKey` errors.
+
+    - parameter name: The name of the missing key
+
+    - returns: A `Decoded.Failure` with a `.MissingKey` error constructed from
```
Comment (klaaspieter) [inline Argo/Types/Decoded/Decoded.swift]: Remove `.`

**Pull request merged**